middleware/validate.js

```diff
@@ -0,0 +1,37 @@
+export function validateArticle(req, res, next) {
+  const { title, content } = req.body;
+  if (!title || !content) {
+    return res.status(400).json({ error: "제목과 내용은 필수입니다." });
+  }
+  next();
+}
+
+export function validateProduct(req, res, next) {
+  const { name, description, price, tags } = req.body;
+  if (!name || !description || price === undefined || !tags) {
+    return res
+      .status(400)
+      .json({ error: "이름, 설명, 가격, 태그는 필수입니다." });
+  }
+  if (typeof price !== "number" || price < 0) {
+    return res.status(400).json({ error: "가격은 0 이상 숫자여야 합니다." });
+  }
```

Comment:
잘 구현되었어요!
하나 추가할 게 있다면, 일반적으로 상품의 가격은 소수점은 아니기 때문에 정수 체크도 추가되면 더 좋겠네요!!

```suggestion
  if (typeof price !== "number" || price < 0 || !Number.isInteger(price)) {
    return res.status(400).json({ error: "가격은 0 이상의 정수여야 합니다." });
  }
```